initgeneration/initgeneration.go

```diff
@@ -0,0 +1,247 @@
+package initgeneration
+
+import (
+	"bytes"
+	"context"
+	_ "embed"
+	"fmt"
+	"text/template"
+
+	dynatracev1alpha1 "github.com/Dynatrace/dynatrace-operator/api/v1alpha1"
+	"github.com/Dynatrace/dynatrace-operator/controllers/kubeobjects"
+	"github.com/Dynatrace/dynatrace-operator/controllers/kubesystem"
+	"github.com/Dynatrace/dynatrace-operator/dtclient"
+	"github.com/Dynatrace/dynatrace-operator/mapper"
+	"github.com/Dynatrace/dynatrace-operator/webhook"
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	notMappedIM              = "-"
+	trustedCASecretField     = "certs"
+	proxyInitSecretField     = "proxy"
+	trustedCAInitSecretField = "ca.pem"
+	initScriptSecretField    = "init.sh"
+)
+
+var (
+	//go:embed init.sh.tmpl
+	scriptContent string
+	scriptTmpl    = template.Must(template.New("initScript").Parse(scriptContent))
+)
+
+// InitGenerator manages the init secret generation for the user namespaces.
+type InitGenerator struct {
+	client    client.Client
+	apiReader client.Reader
+	logger    logr.Logger
+	namespace string
+}
+
+// script holds all the values to be passed to the init script template.
+type script struct {
+	ApiUrl        string
+	SkipCertCheck bool
+	PaaSToken     string
+	Proxy         string
+	TrustedCAs    []byte
+	ClusterID     string
+	TenantUUID    string
+	IMNodes       map[string]string
+}
+
+func NewInitGenerator(client client.Client, apiReader client.Reader, ns string, logger logr.Logger) *InitGenerator {
+	return &InitGenerator{
+		client:    client,
+		apiReader: apiReader,
+		namespace: ns,
+		logger:    logger,
+	}
+}
+
+// GenerateForNamespace creates the init secret for namespace while only having the name of the corresponding dynakube
+// Used by the podInjection webhook in case the namespace lacks the init secret.
+func (g *InitGenerator) GenerateForNamespace(ctx context.Context, dkName, targetNs string) error {
+	g.logger.Info("Reconciling namespace init secret for", "namespace", targetNs)
+	var dk dynatracev1alpha1.DynaKube
+	if err := g.client.Get(context.TODO(), client.ObjectKey{Name: dkName, Namespace: g.namespace}, &dk); err != nil {
+		return err
```

Comment:
Do we want to fail in case we do not find a DynaKube instance or should we silently continue?